Lib/Rabbit4000/tcpip/DMAETH100.LIB

```diff
@@ -808,34 +817,33 @@ dma_rcv_isr::
    jr    nz,.reset_rx_port ; 5
 
    ; Check if only status rcv'd with no packet data
+   ; This is the catch all for errors with R6000
 	ld		hl,(ix+[_decb]+rcvlen)
    test	hl
    jr		z,.test_rx_error
 
    ; Check for valid Rcv Packet
-	ld		a,(ix+[_decb]+rcvlsbstat)
+   ; Don't know where this is documented for R6000?
+#if !defined(_RAB6K)
+   ld		a,(ix+[_decb]+rcvlsbstat)
    bit   7,a               ; 4
    jr    z,.test_rx_error
+#endif
 
 .test_fire_dma:
-   ; Process received packet.
-   ld    a,(ix+[_decb]+receiving)
-   ld    (ix+[_decb]+receiving),0   ; one less available
-
-   ; test whether 'receiving' was decremented to zero...
-   or    a
-   jr    z,.cont20        ; 5 - jump if it was already 0
-   ; yes, receiving was 1, therefore we need to immediately allocate a new rx
-   ; buffer and fire it off.
+   ; Good packet in
+   ld    (ix+[_decb]+receiving),0   ; technically, we're now done with this buffer and will need another
+   ; quickly try to get another buffer
+   call  dmaeth100_alloc_rx
+   jr    c,.cont21				; CF means couldn't get a buffer
 #ifdef DMAETH100_SUPERDEBUG
    ld    l,ISRL_ALLOC_2
    call  write_ilog
 #endif
-   call  dmaeth100_alloc_rx

```

Comment:
I don't think `dmaeth100_alloc_rx` makes use of any registers, so it should be safe to call it after writing `ISRL_ALLOC_2` to the interrupt log.